src/ui/camera.js

```diff
@@ -609,6 +617,113 @@ class Camera extends Evented {
+            selectXAxis ? aabb.min[1] : aabb.max[1],
+            aabb.min[2]
+        ];
+        const axisLength = vec3.length(vec3.sub([], min, max));
+        const minimumDistance = selectXAxis ?
+            axisLength / (2 * Math.tan(tr.fovX * 0.5) * tr.aspect) :
+            axisLength / (2 * Math.tan(tr.fovY * 0.5) * tr.aspect);
+
+        return minimumDistance;
+    }
+
+    _cameraForBoxAndBearingOnGlobe(transform: Transform, p0: LngLatLike, p1: LngLatLike, bearing: number, options?: CameraOptions): ?EasingOptions {
+        const tr = transform.clone();
+        const eOptions = this._extendCameraOptions(options);
+
+        tr.bearing = bearing;
+
+        const coord0 = LngLat.convert(p0);
+        const coord1 = LngLat.convert(p1);
+
+        const midLat = (coord0.lat + coord1.lat) * 0.5;
+        const midLng = (coord0.lng + coord1.lng) * 0.5;
+
+        const origin = latLngToECEF(midLat, midLng);
+
+        const zAxis = vec3.normalize([], origin);
+        const xAxis = vec3.normalize([], vec3.cross([], zAxis, [0, 1, 0]));
+        const yAxis = vec3.cross([], xAxis, zAxis);
+
+        const aabbOrientation = [
+            xAxis[0], xAxis[1], xAxis[2], 0,
+            yAxis[0], yAxis[1], yAxis[2], 0,
+            zAxis[0], zAxis[1], zAxis[2], 0,
+            0, 0, 0, 1
+        ];
+
+        const ecefCoords = [
+            origin,
+
+            latLngToECEF(coord0.lat, coord0.lng),
+            latLngToECEF(coord1.lat, coord0.lng),
+            latLngToECEF(coord1.lat, coord1.lng),
+            latLngToECEF(coord0.lat, coord1.lng),
+
+            latLngToECEF(midLat, coord0.lng),
+            latLngToECEF(midLat, coord1.lng),
+            latLngToECEF(coord0.lat, midLng),
+            latLngToECEF(coord1.lat, midLng),
+        ];
+
+        let aabb = Aabb.fromPoints(ecefCoords.map(p => [vec3.dot(xAxis, p), vec3.dot(yAxis, p), vec3.dot(zAxis, p)]));
+
+        const center = vec3.transformMat4([], aabb.center, aabbOrientation);
+
+        vec3.normalize(center, center);
+        vec3.scale(center, center, GLOBE_RADIUS);
+        tr.center = ecefToLatLng(center);
+
+        const worldToCamera = tr.getWorldToCameraMatrix();
+        const cameraToWorld = mat4.invert(new Float64Array(16), worldToCamera);
+
+        aabb = Aabb.applyTransform(aabb, mat4.multiply([], worldToCamera, aabbOrientation));
+
+        vec3.transformMat4(center, center, worldToCamera);
+
+        const aabbHalfExtentZ = (aabb.max[2] - aabb.min[2]) * 0.5;
+        const frustumDistance = this._minimumAABBFrustumDistance(tr, aabb);
+
+        const offsetZ = vec3.scale([], [0, 0, 1], aabbHalfExtentZ);
+        const aabbClosestPoint = vec3.add(offsetZ, center, offsetZ);
+        const offsetDistance = frustumDistance + (tr.pitch === 0 ? 0 : vec3.distance(center, aabbClosestPoint));
+
+        const globeCenter = tr.globeCenterInViewSpace;
+        const normal = vec3.sub([], center, [globeCenter[0], globeCenter[1], globeCenter[2]]);
```

Comment:
Nit: `globeCenter` is only read here:

```suggestion
        const normal = vec3.sub([], center, globeCenter);
```

Reply:
We might have to leave this one, I just remembered why it was extracted this way:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/ui/camera.js:702:45

Cannot call vec3.sub with globeCenter bound to the third parameter because tuple type [1] is incompatible with array
type [2]. [incompatible-call]

     flow-typed/gl-matrix.js
 [2]   2│ type VecType = Array<number> | Float32Array | Float64Array;

     src/geo/transform.js
 [1] 101│     globeCenterInViewSpace: [number, number, number];
```